libcalico-go/lib/clientv3/tier.go

```diff
@@ -131,7 +152,13 @@ func (r tiers) Delete(ctx context.Context, name string, opts options.DeleteOptio
 func (r tiers) Get(ctx context.Context, name string, opts options.GetOptions) (*apiv3.Tier, error) {
 	out, err := r.client.resources.Get(ctx, opts, apiv3.KindTier, noNamespace, name)
 	if out != nil {
-		return out.(*apiv3.Tier), err
```

Comment:
Sorry, I didn't mean that get would actually update the data in the datastore. Rather, the get would "fill in the field" on the returned object. Here's where we do this for ippools: https://github.com/projectcalico/calico/blob/b5ce29dd4c898cf9b19123a91e6aec21b8f308cc/libcalico-go/lib/clientv3/ippool.go#L244-L287

Reply:
The other important thing is that felix / typha / other components don't use the v3 client, so we need to make sure that they treat "nil" to mean "1,000,000" 

Unfortunately, from a quick skim there doens't seem to be a great centralized place to do that, so we either need to add one or we need to make sure those components that look at tier ordering understand that nil -> 1,000,000.

Considering that this logic only really matters for enterprise, where we have pre-existing tiers, and enterprise only supports KDD mode, we can probably get away with augmenting the Get / List / Watch logic in the k8s backend tier client to populate this field if it is nil: https://github.com/projectcalico/calico/blob/b5ce29dd4c898cf9b19123a91e6aec21b8f308cc/libcalico-go/lib/backend/k8s/resources/tiers.go#L32-L47